conans/test/unittests/model/test_conf.py

```diff
@@ -85,3 +89,65 @@ def test_parse_spaces():
     c = ConfDefinition()
     c.loads(text)
     assert c["core:verbosity"] == "minimal"
+
+
+@pytest.mark.parametrize("text,expected", [
+    (
+        """\
+        # Prepend
+        tools.microsoft.msbuild:verbosity=+another
+        # Append
+        user.company.toolchain:flags+=moreflags
+        """,
+        """\
+        tools.microsoft.msbuild:verbosity=another
+        tools.microsoft.msbuild:verbosity+=minimal
+        user.company.toolchain:flags=someflags
+        user.company.toolchain:flags+=moreflags
+        """
+     ),
+    (
+        """\
+        # Unset
+        tools.microsoft.msbuild:verbosity=!
+        # Define
+        user.company.toolchain:flags=moreflags
+        """,
+        """\
+        tools.microsoft.msbuild:verbosity=!
+        user.company.toolchain:flags=moreflags
+        """
+    )
+])
+def test_conf_actions(conf_definition, text, expected):
+    c, _ = conf_definition
+    text = textwrap.dedent(text)
+    c2 = ConfDefinition()
+    c2.loads(text)
+    c.update_conf_definition(c2)
+    result = textwrap.dedent(expected)
+    assert c.dumps() == result
+
+
+def test_conf_other_patterns_and_access(conf_definition):
+    c, _ = conf_definition
+    text = textwrap.dedent("""\
+        tools.microsoft.msbuild:verbosity=+another
+        tools.microsoft.msbuild:verbosity=!
+        user.company.toolchain:flags=oneflag
+        user.company.toolchain:flags+=secondflag
+        zlib:user.company.toolchain:flags=z1flag
+        zlib:user.company.toolchain:flags+=z2flag
+        openssl:user.company.toolchain:flags=oflag""")
+    c2 = ConfDefinition()
+    c2.loads(text)
+    c.update_conf_definition(c2)
+    c_str = repr(c)
+    assert "None: Conf: OrderedDict([('user.company.toolchain:flags', ConfValues: oneflag secondflag), " \
+           "('tools.microsoft.msbuild:verbosity', ConfValues: )])" in c_str
+    assert "'zlib': Conf: OrderedDict([('user.company.toolchain:flags', ConfValues: z1flag z2flag)])" in c_str
+    assert "'openssl': Conf: OrderedDict([('user.company.toolchain:flags', ConfValues: oflag)])" in c_str
+    assert c["tools.microsoft.msbuild:verbosity"] is None
+    assert c["user.company.toolchain:flags"] == ["oneflag", "secondflag"]
+    assert c["zlib:user.company.toolchain:flags"] == ["z1flag", "z2flag"]
+    assert c["openssl:user.company.toolchain:flags"] == "oflag"  # FIXME: Backward compatibility
```

Comment:
I think this is problematic. That means that every consumer will actually need to check if things are a list, a single value, or None.

We might prefer to normalize this, and we talked to normalize access to the conf (by consumers) with a ``conf.get()`` method. Probably this method can implement the logic:

```python
conf.get("msbuild:verbosity", accept_list=False)  # will raise if the internal list contains more than 1 value, no sense to append or prepend to msbuild:verbosity. 
# we probably want to make ``acept_list=False`` the default (and pick a better name)
conf.get("cmaketoolchain:user_toolchain", accept_list=True) # we know there can be multiple toolchains
```

Reply:
Yes, but I did not want to force it for now. I checked what is already in Conan2.0 and my idea is to bring it back here. Next iteration, I'm going to add that `conf.get()` and I want to discuss if we should get rid of the `[]` access or not for 1.x version.